cpp/src/io/parquet/decode_fixed.cu

```diff
@@ -841,6 +843,33 @@ __device__ inline bool maybe_has_nulls(page_state_s* s)
   return run_val != s->col.max_level[lvl];
 }
 
+template <int decode_block_size_t, typename state_buf>
+inline __device__ void bool_plain_decode(page_state_s* s, state_buf* sb, int t, int to_decode)
+{
+  int pos              = s->dict_pos;
+  int const target_pos = pos + to_decode;
+
+  while (pos < target_pos) {
+    int const batch_len = min(target_pos - pos, decode_block_size_t);
+
+    if (t < batch_len) {
+      int const bit_pos           = pos + t;
+      int const byte_offset       = bit_pos >> 3;
+      int const bit_in_byte_index = bit_pos & 7;
+
+      uint8_t const* const read_from = s->data_start + byte_offset;
+      bool const read_bit            = *read_from & (1 << bit_in_byte_index);
+
+      int const write_to_index     = rolling_index<state_buf::dict_buf_size>(bit_pos);
+      sb->dict_idx[write_to_index] = read_bit;
```

Comment:
Future thought:  `dict_idx` might benefit from a rename since it gets repurposed like this from time to time.

Reply:
Yeah, I've found it really difficult to parse what's going on with all of the variable repurposing.  Definitely worth looking into how to clean some of this up once we're all done.